### ISSUE TITLE
Problem using LanguageServer: LoadError: UndefVarError: SymbolServerProcess not defined

### ISSUE THREAD

**Issue by arrufat** [JuliaEditorSupport/LanguageServer.jl#300]
Hi, I'm trying to setup the LanguageServer with Julia 1.0.1, but I'm getting this error:
``` julia
]add LanguageServer#master
...
using LanguageServer
...
ERROR: LoadError: LoadError: UndefVarError: SymbolServerProcess not defined
Stacktrace:
 [1] top-level scope at none:0
 [2] include at ./boot.jl:317 [inlined]
 [3] include_relative(::Module, ::String) at ./loading.jl:1041
 [4] include at ./sysimg.jl:29 [inlined]
 [5] include(::String) at /home/adria/.julia/packages/LanguageServer/C1B4E/src/LanguageServer.jl:1
 [6] top-level scope at none:0
 [7] top-level scope at none:2
in expression starting at /home/adria/.julia/packages/LanguageServer/C1B4E/src/languageserverinstance.jl:3
in expression starting at /home/adria/.julia/packages/LanguageServer/C1B4E/src/LanguageServer.jl:17
ERROR: Failed to precompile LanguageServer [2b0e0bc5-e4fd-59b4-8912-456d1b03d8d7] to /home/adria/.julia/compiled/v1.0/LanguageServer/ite7n.ji.
```
Any ideas? Thanks in advance.

**Comment by jingpengw** [comment]
SymbolServer.jl was not registered, so you need to install it manually.
`add https://github.com/JuliaEditorSupport/SymbolServer.jl`

**Comment by dileep-kishore** [comment]
That did not fix the problem.

**Comment by rishisubramanian** [comment]
I'm having the same issue. Julia 1.0.1 on Arch. tried installing LanguageServer from master. Installing SymbolServer didn't work.

**Comment by jingpengw** [comment]
me too, seems like we need this PR passing the tests.
https://github.com/JuliaEditorSupport/LanguageServer.jl/pull/290

**Comment by benneti** [comment]
It still does not work here (Arch Linux, too, Julia 1.0.2)

```
pkg> add https://github.com/JuliaEditorSupport/SymbolServer.jl
pkg> add LanguageServer#master
julia> using LanguageServer
```

Yields the same error.

```
julia>  using SymbolServer
julia> s = SymbolServerProcess()
julia> getstore(s)
```
results in 
```
WARNING: Method definition (::Type{StaticLint.ImportBinding})(Any, Any, Any, Any) in module StaticLint at ~/.julia/packages/StaticLint/LsaCh/src/bindings.jl:2 overwritten at ~/.julia/packages/StaticLint/LsaCh/src/bindings.jl:7.
WARNING: Method definition (::Type{StaticLint.Binding})(Any, Any, Any, Any, Any) in module StaticLint at ~/.julia/packages/StaticLint/LsaCh/src/bindings.jl:10 overwritten at ~/.julia/packages/StaticLint/LsaCh/src/bindings.jl:16.
WARNING: Method definition (::Type{StaticLint.ImportBinding})(Any, Any, Any, Any) in module StaticLint at ~/.julia/packages/StaticLint/LsaCh/src/bindings.jl:7 overwritten at ~/.julia/packages/StaticLint/LsaCh/src/bindings.jl:2.
WARNING: Method definition (::Type{StaticLint.ImportBinding})(Any, Any, Any, Any) in module StaticLint at ~/.julia/packages/StaticLint/LsaCh/src/bindings.jl:2 overwritten at ~/.julia/packages/StaticLint/LsaCh/src/bindings.jl:7.
WARNING: Method definition (::Type{StaticLint.Binding})(Any, Any, Any, Any, Any) in module StaticLint at ~/.julia/packages/StaticLint/LsaCh/src/bindings.jl:16 overwritten at ~/.julia/packages/StaticLint/LsaCh/src/bindings.jl:10.
WARNING: Method definition (::Type{StaticLint.Binding})(Any, Any, Any, Any, Any) in module StaticLint at ~/.julia/packages/StaticLint/LsaCh/src/bindings.jl:10 overwritten at ~/.julia/packages/StaticLint/LsaCh/src/bindings.jl:16.
ERROR: LoadError: LoadError: UndefVarError: SymbolServerProcess not defined
Stacktrace:
 [1] top-level scope at none:0
 [2] include at ./boot.jl:317 [inlined]
 [3] include_relative(::Module, ::String) at ./loading.jl:1044
 [4] include at ./sysimg.jl:29 [inlined]
 [5] include(::String) at ~/.julia/packages/LanguageServer/goTQ0/src/LanguageServer.jl:1
 [6] top-level scope at none:0
 [7] top-level scope at none:2
in expression starting at ~/.julia/packages/LanguageServer/goTQ0/src/languageserverinstance.jl:3
in expression starting at ~/.julia/packages/LanguageServer/goTQ0/src/LanguageServer.jl:18
Dict{String,Any} with 4 entries:
  "Core"           => ModuleStore("Core", Dict{String,Any}("UndefInitializer"=>primitiveStore(String[], "```\nUndefInitializer\n```\n\nSingleton type used in array initializ…
  "Base"           => ModuleStore("Base", Dict{String,Any}("@time"=>FunctionStore(SymbolServer.MethodStore[MethodStore("/usr/bin/../share/julia/base/util.jl", 153, Tuple{Str…
  "LanguageServer" => ModuleStore("LanguageServer", Dict{String,Any}(), Set(String[]), "")
  "SymbolServer"   => ModuleStore("SymbolServer", Dict{String,Any}(), Set(String[]), "")
```

Starting julia using `bash -c julia` and then executing the above only yields
```
Dict{String,Any} with 4 entries:
  "Core"           => ModuleStore("Core", Dict{String,Any}("UndefInitializer"=>primitiveStore(String[], "```\nUndefInitializer\n```\n\nSingleton type used in array initializ…
  "Base"           => ModuleStore("Base", Dict{String,Any}("@time"=>FunctionStore(SymbolServer.MethodStore[MethodStore("/usr/bin/../share/julia/base/util.jl", 153, Tuple{Str…
  "LanguageServer" => ModuleStore("LanguageServer", Dict{String,Any}(), Set(String[]), "")
  "SymbolServer"   => ModuleStore("SymbolServer", Dict{String,Any}(), Set(String[]), "")

```

But `using LanguageServer` won't work either

**Comment by arrufat** [comment]
@benneti I can reproduce your problem exactly

**Comment by benneti** [comment]
@rishisubramanian 's solution makes it possible to `using LanguageServer` for me, too

**Comment by davidanthoff** [comment]
This should have been fixed for quite a while now on ``master``. Is this still a problem for folks?

**Comment by arrufat** [comment]
@davidanthoff I'm able to build it without issues, however, I can't use the languageserver in neither vscode nor vim with julia 1.0.2 on Arch Linux. Is this a known issue ?

**Comment by davidanthoff** [comment]
Language server should work in VS Code, if that doesn't work, it would be great if you could open a new issue for that! One more clarifying question: when you say it doesn't work in VS Code, do you mean that the beta releases we are putting out aren't working on your system?

**Comment by rishisubramanian** [comment]
It works for me in Emacs. Can you run the LanguageServer command manually? One thing that changed recently is that SymbolServer has to be imported along with LanguageServer and I'm not sure if VSCode and Vim have updated their plugins.

**Comment by davidanthoff** [comment]
If you grab betas of the VS Code extension from [here](https://github.com/JuliaEditorSupport/julia-vscode/releases), it should all work.

**Comment by arrufat** [comment]
@davidanthoff Yes, I confirm completion works with the beta extensions. Thank you very much!

**Comment by ffevotte** [comment]
I'm still having the exact same issue as the OP in the first post:

```sh
$ julia --version
julia version 1.0.3
```

```
(v1.0) pkg> add LanguageServer#master
   Cloning default registries into /home/francois/.julia/registries
   Cloning registry General from "https://github.com/JuliaRegistries/General.git"
   Cloning git-repo `https://github.com/JuliaEditorSupport/LanguageServer.jl.git`
  Updating git-repo `https://github.com/JuliaEditorSupport/LanguageServer.jl.git`
 Resolving package versions...
 Installed URIParser ────── v0.4.0
 Installed SymbolServer ─── v0.1.0
 Installed DocumentFormat ─ v0.2.0
 Installed StaticLint ───── v0.0.2
 Installed Tokenize ─────── v0.5.3
 Installed JSON ─────────── v0.20.0
 Installed CSTParser ────── v0.4.1
```

```
julia> using LanguageServer
[ Info: Precompiling LanguageServer [2b0e0bc5-e4fd-59b4-8912-456d1b03d8d7]
...
ERROR: LoadError: LoadError: UndefVarError: SymbolServerProcess not defined
Stacktrace:
 [1] getproperty(::Module, ::Symbol) at ./sysimg.jl:13
 [2] top-level scope at none:0
 [3] include at ./boot.jl:317 [inlined]
 [4] include_relative(::Module, ::String) at ./loading.jl:1044
 [5] include at ./sysimg.jl:29 [inlined]
 [6] include(::String) at /home/francois/.julia/packages/LanguageServer/UHqIL/src/LanguageServer.jl:1
 [7] top-level scope at none:0
 [8] include at ./boot.jl:317 [inlined]
 [9] include_relative(::Module, ::String) at ./loading.jl:1044
 [10] include(::Module, ::String) at ./sysimg.jl:29
 [11] top-level scope at none:2
 [12] eval at ./boot.jl:319 [inlined]
 [13] eval(::Expr) at ./client.jl:393
 [14] top-level scope at ./none:3
in expression starting at .../LanguageServer/UHqIL/src/languageserverinstance.jl:3
in expression starting at .../LanguageServer/UHqIL/src/LanguageServer.jl:18
```

The solution proposed by @rishisubramanian above seems to get me a bit further, but I hit another error with `StaticLint.LintError` in `src/parsing.jl:51`.

Do I have a bad version of `StaticLint`? Or should I open a new issue for this error?

**Comment by ffevotte** [comment]
I don't manage to have the master of StaticLint at the same time as LanguageServer: there seem to be incompatibilities between CSTParser, StaticLint and DocumentFormat.

Starting from a fresh Julia install:

```
(v1.0) pkg> add LanguageServer#master
   Cloning default registries into /home/francois/.julia/registries
   Cloning registry General from "https://github.com/JuliaRegistries/General.git"
   Cloning git-repo `https://github.com/JuliaEditorSupport/LanguageServer.jl.git`
  Updating git-repo `https://github.com/JuliaEditorSupport/LanguageServer.jl.git`
 Resolving package versions...
 Installed URIParser ────── v0.4.0
 Installed DocumentFormat ─ v0.2.0
 Installed Tokenize ─────── v0.5.3
 Installed SymbolServer ─── v0.1.0
 Installed StaticLint ───── v0.0.2
 Installed JSON ─────────── v0.20.0
 Installed CSTParser ────── v0.4.1
...

(v1.0) pkg> add SymbolServer#master
   Cloning git-repo `https://github.com/JuliaEditorSupport/SymbolServer.jl.git`
  Updating git-repo `https://github.com/JuliaEditorSupport/SymbolServer.jl.git`
 Resolving package versions...
  Updating `~/.julia/environments/v1.0/Project.toml`
  [cf896787] + SymbolServer v0.1.0+ #master (https://github.com/JuliaEditorSupport/SymbolServer.jl.git)
  Updating `~/.julia/environments/v1.0/Manifest.toml`
  [cf896787] ↑ SymbolServer v0.1.0 ⇒ v0.1.0+ #master (https://github.com/JuliaEditorSupport/SymbolServer.jl.git)

(v1.0) pkg> add StaticLint#master
   Cloning git-repo `https://github.com/ZacLN/StaticLint.jl.git`
  Updating git-repo `https://github.com/ZacLN/StaticLint.jl.git`
 Resolving package versions...
ERROR: Unsatisfiable requirements detected for package DocumentFormat [ffa9a821]:
 DocumentFormat [ffa9a821] log:
 ├─possible versions are: [0.0.1, 0.1.0, 0.2.0] or uninstalled
 ├─restricted to versions 0.2.0-* by LanguageServer [2b0e0bc5], leaving only versions 0.2.0
 │ └─LanguageServer [2b0e0bc5] log:
 │   ├─possible versions are: 0.5.0 or uninstalled
 │   └─LanguageServer [2b0e0bc5] is fixed to version 0.5.0+
 └─restricted by compatibility requirements with CSTParser [00ebfdb7] to versions: uninstalled — no versions left
   └─CSTParser [00ebfdb7] log:
     ├─possible versions are: [0.0.2-0.0.6, 0.1.0-0.1.1, 0.2.0-0.2.2, 0.3.0-0.3.5, 0.4.0-0.4.1, 0.5.0-0.5.1] or uninstalled
     └─restricted to versions 0.5.0-* by StaticLint [b3cc710f], leaving only versions 0.5.0-0.5.1
       └─StaticLint [b3cc710f] log:
         ├─possible versions are: 0.1.0 or uninstalled
         └─StaticLint [b3cc710f] is fixed to version 0.1.0+
```

**Comment by ZacLN** [comment]
Can you use `]dev LanguageServer StaticLint SymbolServer` to pull master of each package?

**Comment by ffevotte** [comment]
Yes but I get the same error:

```
$ rm -rf ~/.julia
$ julia
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.0.3 (2018-12-18)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

(v1.0) pkg> dev LanguageServer StaticLint SymbolServer
   Cloning default registries into /home/francois/.julia/registries
   Cloning registry General from "https://github.com/JuliaRegistries/General.git"
  Updating registry at `~/.julia/registries/General`
  Updating git-repo `https://github.com/JuliaRegistries/General.git`
   Cloning git-repo `https://github.com/JuliaEditorSupport/LanguageServer.jl.git`
  Updating git-repo `https://github.com/JuliaEditorSupport/LanguageServer.jl.git`
   Cloning git-repo `https://github.com/ZacLN/StaticLint.jl.git`
  Updating git-repo `https://github.com/ZacLN/StaticLint.jl.git`
   Cloning git-repo `https://github.com/JuliaEditorSupport/SymbolServer.jl.git`
  Updating git-repo `https://github.com/JuliaEditorSupport/SymbolServer.jl.git`
 Resolving package versions...
ERROR: Unsatisfiable requirements detected for package DocumentFormat [ffa9a821]:
 DocumentFormat [ffa9a821] log:
 ├─possible versions are: [0.0.1, 0.1.0, 0.2.0] or uninstalled
 ├─restricted to versions 0.2.0-* by LanguageServer [2b0e0bc5], leaving only versions 0.2.0
 │ └─LanguageServer [2b0e0bc5] log:
 │   ├─possible versions are: 0.5.0 or uninstalled
 │   └─LanguageServer [2b0e0bc5] is fixed to version 0.5.0+
 └─restricted by compatibility requirements with CSTParser [00ebfdb7] to versions: uninstalled — no versions left
   └─CSTParser [00ebfdb7] log:
     ├─possible versions are: [0.0.2-0.0.6, 0.1.0-0.1.1, 0.2.0-0.2.2, 0.3.0-0.3.5, 0.4.0-0.4.1, 0.5.0-0.5.1] or uninstalled
     └─restricted to versions 0.5.0-* by StaticLint [b3cc710f], leaving only versions 0.5.0-0.5.1
       └─StaticLint [b3cc710f] log:
         ├─possible versions are: 0.1.0 or uninstalled
         └─StaticLint [b3cc710f] is fixed to version 0.1.0+
```

**Comment by ZacLN** [comment]
How about either:
1. uninstalling the packages then: `]dev LanguageServer StaticLint SymbolServer DocumentFormat` (and CSTParser/Tokenize if necessary); or
2. cloning the packages directly from the repos `]dev https://github.com/JuliaEditorSupport/LanguageServer.jl etc...`

I'm not sure why this is so muddled up I'm afraid

**Comment by ffevotte** [comment]
OK, it looks like DocumentFormat is the one which really needs to be pulled from master in order to avoid dependency problems.

This seems to work:
```
(v1.0) pkg> add LanguageServer#master StaticLint#master DocumentFormat#master SymbolServer#master
   Cloning default registries into /home/francois/.julia/registries
   Cloning registry General from "https://github.com/JuliaRegistries/General.git"
   Cloning git-repo `https://github.com/JuliaEditorSupport/LanguageServer.jl.git`
  Updating git-repo `https://github.com/JuliaEditorSupport/LanguageServer.jl.git`
   Cloning git-repo `https://github.com/ZacLN/StaticLint.jl.git`
  Updating git-repo `https://github.com/ZacLN/StaticLint.jl.git`
   Cloning git-repo `https://github.com/ZacLN/DocumentFormat.jl.git`
  Updating git-repo `https://github.com/ZacLN/DocumentFormat.jl.git`
   Cloning git-repo `https://github.com/JuliaEditorSupport/SymbolServer.jl.git`
  Updating git-repo `https://github.com/JuliaEditorSupport/SymbolServer.jl.git`
 Resolving package versions...
 Installed URIParser ─ v0.4.0
 Installed Tokenize ── v0.5.3
 Installed CSTParser ─ v0.5.1
 Installed JSON ────── v0.20.0
...
```

Then I seem to be able to run a server manually without problems:
```
using LanguageServer
using SymbolServer
server = LanguageServer.LanguageServerInstance(stdin, stdout, false, "/home/francois/.julia/environments/v1.0", "", Dict())
server.runlinter = true
run(server)
```

Thanks for your help!

**Comment by davidanthoff** [comment]
DocumentFormat.jl latest tag hasn't made it through METADATA, see [here](https://github.com/JuliaLang/METADATA.jl/pull/20306). I assume that once that PR is merged, everything should "just work", i.e. a simple ``]add LanguageServer`` should be enough.

**Comment by jamm1985** [comment]
it works if `add DocumentFormat#master`

**Comment by davidanthoff** [comment]
Hopefully this will be resolved once https://github.com/JuliaLang/METADATA.jl/pull/20556 is merged.

**Comment by ikselven** [comment]
Ran into this issue, too, and it's resolved now. Thanks!
